persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/StargateQueryHandler.java

```diff
@@ -145,17 +145,17 @@ public ResultMessage process(
       QueryProcessor.metrics.regularStatementsExecuted.inc();
     }
 
+    if (customPayload != null && customPayload.containsKey("stargate.auth.subject.token")) {
+      authorizeByToken(customPayload, statement);
+    }
```

Comment:
Since we're tightening security in this PR, I guess we ought to check whether the token is _expected_ to be in the payload. Right now we allow the statement to proceed if the token is _not_ present... perhaps it is not present when if _should_ be present? Can we detect that (and fail in that case)? WDYT?

Tangential: It might be worth moving the `if` into `authorizeByToken` in case the decision is going to involve other conditions.

Reply:
Well, you can have a customPayload without using token auth. Plus even if we set `-Dstargate.auth_api_enable_username_token=true -Dstargate.cql_use_auth_service=true` this could still be a login with a username/password so there really isn't a good way to know if it _should_ be there unless I'm missing something.